src/sentry/api/endpoints/organization_api_key_details.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import absolute_import
+
+from rest_framework import serializers, status
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationApiKeysPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.models import ApiKey, AuditLogEntryEvent
+
+
+class ApiKeySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ApiKey
+        fields = ('label', 'scope_list', 'allowed_origins')
+
+
+class OrganizationApiKeyDetailsEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationApiKeysPermission, )
+
+    def get(self, request, organization, api_key_id):
+        try:
+            apikey = ApiKey.objects.get(id=api_key_id,
+                                        organization_id=organization.id,
+                                        )
+        except ApiKey.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return Response(serialize(apikey, request.user))
+
+    def put(self, request, organization, api_key_id):
+        """
+        Update an API Key
+        `````````````
+
+        :pparam string organization_slug: the slug of the organization the
+                                          team belongs to.
+        :pparam string api_key_id: the api key id
```

Comment:
i think we generally don't use the numerical ids of objects for endpoints, so you may want to use the `key` property as the identifier instead

Reply:
In production we're using the numeric value, should we change it to use the key?

Reply:
Hmm, I guess I was thinking about it more along the lines of if we're going to add this to the public API docs (either now or in the future), it might be nice if it were a more human readable identifier. For example: for projects, teams, and organizations, we use the slug rather than the id, and for releases, we use version rather than the id. But, it looks like we use the numerical id for groups (rather than something like the short id) in publicly documented endpoints, so it probably doesn't matter after all.